test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointLivenessUpdaterUnitTest.java

```diff
@@ -66,4 +72,34 @@ public void testUpdateLiveness() {
         verify(txn).putRecord(Matchers.any(), Matchers.any(), captor.capture(), Matchers.any());
         Assert.assertEquals(1, captor.getValue().getSyncHeartbeat());
     }
+
+    @RepeatedTest(NUM_REPETITIONS)
+    public void testUpdateLivenessRaceCondition() {
+        ActiveCPStreamMsg activeCPStreamMsg = ActiveCPStreamMsg.newBuilder().setSyncHeartbeat(0).build();
+        when((ActiveCPStreamMsg) corfuStoreEntry.getPayload()).thenReturn(activeCPStreamMsg);
+
+        //This block helps with possibly triggering a race condition
+        long endTime = System.nanoTime() + TimeUnit.NANOSECONDS.convert(LONG_INTERVAL, TimeUnit.SECONDS);
+        boolean clear = true;
+        while (System.nanoTime() < endTime) {
+            if (clear) {
+                livenessUpdater.notifyOnSyncComplete();
+            } else {
+                livenessUpdater.updateLiveness(tableName);
+            }
+            clear = !clear;
+        }
+
+        //ensures atLeastOnce invocation of putRecord if there were no exceptions in updateHeartbeat method
+        livenessUpdater.updateLiveness(tableName);
+        try {
+            TimeUnit.SECONDS.sleep(INTERVAL.getSeconds());
+        } catch (InterruptedException e) {
+            log.warn("Sleep interrupted: ", e);
+        }
+        livenessUpdater.notifyOnSyncComplete();
+
+        ArgumentCaptor<ActiveCPStreamMsg> captor = ArgumentCaptor.forClass(ActiveCPStreamMsg.class);
+        verify(txn, atLeastOnce()).putRecord(Matchers.any(), Matchers.any(), captor.capture(), Matchers.any());
```

Comment:
It doesn't seem like this argument captor is being used. Did you mean to validate anything here? If not, Matchers.any() can be used instead.

Reply:
Updated it.